ocaml/xapi/xapi_pool.ml

```diff
@@ -3345,10 +3346,30 @@ let enable_tls_verification ~__context =
   | Some self ->
       Xapi_cluster_host.set_tls_config ~__context ~self ~verify:true
 
+let contains_bundle_repo ~__context ~repos =
+  List.exists
+    (fun repo -> Db.Repository.get_origin ~__context ~self:repo = `bundle)
+    repos
+
+let assert_enable_single_bundle_repo ~__context ~repos =
+  if List.length repos > 1 && contains_bundle_repo ~__context ~repos then
+    raise Api_errors.(Server_error (bundle_repo_should_be_single_enabled, []))
+
+let assert_not_sync_bundle_repo ~__context ~repos =
+  if contains_bundle_repo ~__context ~repos then
+    raise Api_errors.(Server_error (can_not_sync_updates, []))
+
+let disable_auto_update_sync_for_bundle_repo ~__context ~self ~repos =
+  if contains_bundle_repo ~__context ~repos then (
+    Pool_periodic_update_sync.set_enabled ~__context ~value:false ;
+    Db.Pool.set_update_sync_enabled ~__context ~self ~value:false
+  )
+
 let set_repositories ~__context ~self ~value =
   Xapi_pool_helpers.with_pool_operation ~__context ~self
     ~doc:"pool.set_repositories" ~op:`configure_repositories
   @@ fun () ->
+  assert_enable_single_bundle_repo ~__context ~repos:value ;
```

Comment:
This function name is a little confusing, how about `assert_single_bundle_repo_can_be_enabled`